client/dashboard/me/billing-purchases/cancel-purchase/cancel-purchase-form/step-components/feedback-step.tsx

```diff
@@ -136,32 +136,35 @@ function ImportQuestion( { onChange }: { onChange?: ChangeCallback } ) {
 
 type FeedbackStepProps = {
 	purchase: Purchase;
+	plans: PlanProduct[];
 	isImport: boolean;
 	cancellationReasonCodes: string[];
 	onChangeCancellationReason: ChangeCallback;
 	onChangeCancellationReasonDetails: ChangeCallback;
-	onChangeImportFeedback?: ChangeCallback;
+	onChangeImportFeedback: ChangeCallback;
 };
 
 export default function FeedbackStep( {
 	purchase,
+	plans,
 	isImport,
 	cancellationReasonCodes,
 	onChangeCancellationReason,
 	onChangeCancellationReasonDetails,
 	onChangeImportFeedback,
 }: FeedbackStepProps ) {
-	const isPlanPurchase = isPlan( purchase );
-	const isGSuite = isGSuiteOrGoogleWorkspace( purchase );
-	const isDomain = isDomainRegistration( purchase );
+	const isPlanPurchase = purchase.is_plan;
+	const isGSuite = purchase.is_google_workspace_product;
+	const isDomain = purchase.is_domain_registration;
 	const showCancellationReason = isPlanPurchase || isGSuite || isDomain;
 
 	return (
 		<>
 			{ showCancellationReason && (
 				<CancellationReason
+					plans={ plans }
 					purchase={ purchase }
-					reasonCodes={ cancellationReasonCodes }
+					reasonCodes={ cancellationReasonCodes ?? [] }
```

Comment:
It looks like `cancellationReasonCodes` is required so this would not be needed. Is there a chance it could be undefined?

Reply:
Great catch! I went back and forth on this a bit and need to remove this. Previously I made it optional, but now it's required.